### PR TITLE
Add waist_imu in urdf file

### PR DESCRIPTION
**Pull request by martinaxgloria**
As per title. WIP

### CODE REVIEW

**Comment by mfussi66** [comment]
Thanks @martinaxgloria ! Could you add in the PR post a test in the simulator?

File: urdf/ergoCub/robots/ergoCubGazeboV1_minContacts/model.urdf [hunk 1]
```diff
@@ -3291,6 +3298,20 @@
     <parent link="head"/>
     <origin rpy="3.141592653589793 -0.0 1.5707963267948966" xyz="-0.004299998 -0.0055 0.03823499999999996"/>
   </sensor>
+  <gazebo reference="torso_1">
+    <sensor name="waist_imu_0" type="imu">
+      <always_on>1</always_on>
+      <update_rate>100</update_rate>
+      <pose>0.0354497 0.00639688 0.060600 -1.5707322 1.5707963 0</pose>
+      <plugin name="ergocub_yarp_gazebo_plugin_IMU" filename="libgazebo_yarp_imu.so">
+    <yarpConfigurationFile>model://ergoCub/conf/gazebo_ergocub_inertial.ini</yarpConfigurationFile>
```
Comment (mfussi66) [inline urdf/ergoCub/robots/ergoCubGazeboV1_minContacts/model.urdf:3307]: .ini file should be gazebo_ergocub_waist_inertial.ini

Comment (martinaxgloria) [reply urdf/ergoCub/robots/ergoCubGazeboV1_minContacts/model.urdf]: Done here https://github.com/icub-tech-iit/ergocub-software/pull/99/commits/72a2fda9265cfaf6f5ea467af2cb0dac03d755eb

File: urdf/ergoCub/robots/ergoCubGazeboV1/model.urdf [hunk 2]
```diff
@@ -3291,6 +3298,20 @@
     <parent link="head"/>
     <origin rpy="3.141592653589793 -0.0 1.5707963267948966" xyz="-0.004299998 -0.0055 0.03823499999999996"/>
   </sensor>
+  <gazebo reference="torso_1">
+    <sensor name="waist_imu_0" type="imu">
+      <always_on>1</always_on>
+      <update_rate>100</update_rate>
+      <pose>0.0354497 0.00639688 0.060600 -1.5707322 1.5707963 0</pose>
+      <plugin name="ergocub_yarp_gazebo_plugin_IMU" filename="libgazebo_yarp_imu.so">
+    <yarpConfigurationFile>model://ergoCub/conf/gazebo_ergocub_inertial.ini</yarpConfigurationFile>
```
[the remark this answers is not in the document]
Comment (martinaxgloria) [reply urdf/ergoCub/robots/ergoCubGazeboV1/model.urdf]: Done here https://github.com/icub-tech-iit/ergocub-software/pull/99/commits/72a2fda9265cfaf6f5ea467af2cb0dac03d755eb

**Review by Nicogene — CHANGES_REQUESTED** [review]
The changes made on the urdf have been done by hand or it is the result of the generation from `simmechanics_to_urdf`?

Checking the CI the generation of the urdf fails because `XSENS_MTI_630_IMU_CSYS` does not exist in the simmechanics xml, since we decided to not update for now the simulation model with the creo 9->7 procedure.

The only idea I have is to refer to the torso_1 csys and use 

`additionalTransformation` -> https://github.com/robotology/icub-models-generator/blob/fd2d430a399119f2ded5a15f1c8ea4185b9aca6c/simmechanics/CMakeLists.txt#L279

@traversaro is there a better way to add a frame that does not exist in the xml?

**Comment by martinaxgloria** [comment]
> The changes made on the urdf have been done by hand or it is the result of the generation from `simmechanics_to_urdf`?

I did it by hand.

**Comment by martinaxgloria** [comment]
I added the `sensorBlob` for waist_imu here: https://github.com/icub-tech-iit/ergocub-software/pull/99/files#diff-d43fa5cafdc7096f7b13d331fdfa24223268f048fc00d8c05648fa95e7a3832b

@traversaro is this what you were referring to?

**Comment by martinaxgloria** [comment]
Here https://github.com/icub-tech-iit/ergocub-software/pull/99/commits/47900b1650a8d5a57f8f7a321090f0d6bf568c8d I added the XMLBlobs as suggested. Moreover, I restore the urdf file, so that it is not modified by hand anymore, and I update the mass and inertia properties in the .yaml file

@Nicogene

**Pull request merged**